iaf-commons/src/main/java/nl/nn/adapterframework/util/YamlParser.java

```diff
@@ -0,0 +1,114 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.util;
+
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.yaml.snakeyaml.Yaml;
+
+public class YamlParser extends Properties {
```

Comment:
I don't think it is very good design that the parser-class is also the class that stores the data, I would decouple that.

Especially I would decouple the loading / parsing from the constructor.

Reply:
Due the code being recursive I strongly suggest not to do this...

Reply:
I do see the code is recursive but I think you can still recursively load the data into a map or Properties object that is passed as parameter, instead of being a member?